### PR TITLE
allow specifiying custom image url

### PR DESCRIPTION
**Pull request by epsilon-0**
Signed-off-by: Aisha Tammy <floss@bsd.ac>

First time working on a github action, so may have made some mistakes.
This is in conjunction with the PR to build OpenBSD snapshot images on the openbsd-builder.

### CODE REVIEW

File: src/action/input.ts [hunk 1]
```diff
@@ -18,6 +19,13 @@ export class Input {
     }))
   }
 
+  get image_url(): string {
```
Comment (jacob-carlborg) [inline src/action/input.ts:22]: Please use camel case.

Comment (jacob-carlborg) [inline src/action/input.ts:22]: All inputs need to be added to [`action.yml`](https://github.com/cross-platform-actions/action/blob/master/action.yml).

**Comment by jacob-carlborg** [comment]
There needs to be a test. I probably need to create a new repository containing a VM image.

**Comment by jacob-carlborg** [comment]
Since this is TypeScript, the code needs to be compiled to JavaScript. You need to run `npm run all` and commit all changed files.

**Review by jacob-carlborg — CHANGES_REQUESTED** [review]
See comments.

File: src/action/input.ts (hunk 1 above)
Comment (epsilon-0) [reply src/action/input.ts]: done.

Comment (epsilon-0) [reply src/action/input.ts]: done.

**Comment by epsilon-0** [comment]
Addressed all the comments and have updated the PR.

**Comment by jacob-carlborg** [comment]
Please update the readme and the changelog as well.

**Comment by epsilon-0** [comment]
added

File: src/action/action.ts [hunk 2]
```diff
@@ -124,12 +124,9 @@ export class Action {
   }
 
   async downloadDiskImage(): Promise<string> {
-    core.info(
-      `Downloading disk image: ${this.operatingSystem.virtualMachineImageUrl}`
-    )
-    const result = await cache.downloadTool(
-      this.operatingSystem.virtualMachineImageUrl
-    )
+    const imageURL = `this.input.imageURL !== '' ? this.input.imageURL : this.operatingSystem.virtualMachineImageUrl`
```
Comment (jacob-carlborg) [inline src/action/action.ts:127]: This doesn't look right. The whole expression on the right hand side is a string.

Comment (epsilon-0) [reply src/action/action.ts]: oops, yes you are right. 
Fixed.

**Comment by jacob-carlborg** [comment]
I've created a new repository for a custom VM image [1]. Could you please add a test for that?

[1] https://github.com/cross-platform-actions/test-custom-image-builder/releases/tag/v1.0.0

**Comment by jacob-carlborg** [comment]
Merged in: https://github.com/cross-platform-actions/action/commit/12af1e9f176a6e7df6737a58f5eac8bede3adb46.

**Pull request closed without merging**